text/0010-bft-signatures.md

```diff
@@ -0,0 +1,291 @@
+
+To that end, SmartBFT substitutes the identity (over 700B in size) with its consensus node identifier (a 64 bit number). 
+As a result, the `MetadataSignature` now contains three additional fields:
+
+```
+message MetadataSignature {
+    bytes signature_header = 1; // An encoded SignatureHeader
+    bytes signature = 2;       // The signature over the concatenation of the Metadata value bytes, signatureHeader, and block header
+    uint64 signer_id = 3;
+    bytes nonce = 4;
+}
+```
+
+When a Fabric node validates a signature on a block, an ephemeral `SignatureHeader` is constructed where the `nonce` is taken from the
+`MetadataSignature` and the `creator` is the `identity` matching the `signer_id`. 
+
+This technique reduces the aforementioned overhead of 3.5KB to a meager 40 bytes.
+
+
+While the approach of the SmartBFT library integration works, it creates a coupling between the consensus implementation and the rest of Fabric, and this limits incorporation of other consensus implementations into Fabric.
+ 
+More specifically, it requires peers to understand how to decode the consensus metadata of ordering service nodes. That means that the peer needs to be aware of the type of ordering service it is working with, which breaks the abstraction of the ordering service as a service that receives as input transactions, and outputs blocks.
+
+
+# Proposed techniques
+[technique]: #proposed-technique
+
+The proposed approach is heavily inspired by SmartBFT but is designed to be consensus agnostic, minimize implementation and operational complexity.
+
+### Block replication
+
+The proposed technique for block replication is essentially the approach of SmartBFT where a re-shuffling takes place not only upon suspicion of the block source but also in a timely fashion in order to be resilient to overloaded or deliberately slow orderers.   
+In order to save network bandwidth and processing power when the network is idle, the automatic time-based re-shuffling will only take place if new blocks are received.
+Moreover, there will not be a **need** to specify in the peer's local configuration (or an orderer's local configuration) which of the implementations to use. 
+Notice that a block validation policy that mandates signatures from multiple ordering nodes, is **never** satisfied by **only** principal sets of size one. 
+Or, equivalently, a block validation policy that is satisfied by **only** principal sets of size one never requires signatures from multiple ordering nodes. 
+Evidently, policies of the former type are **always** BFT policies, while policies of the latter type might be either a crash fault policy or a BFT policy that utilizes a threshold signature scheme.
+
+Therefore, Fabric nodes will pick the BFT implementation if either the block validation policy implies the need, or if an override is activated in the node's local configuration.  
```

Comment:
The above para gives an impression that you are proposing this technique of connecting with `f+1` as a general mechanism so as to avoid the need of a local configuration. But in the last statement, you mention `if an override is activated in the node's local configuration`. Are you proposing a local configuration here? Also, it's worth mentioning if the peer is expected to derive the value of `f` (for enable connecting to `f+1`) from the total numbers of OSNs mentioned in the channel config.

Reply:
It is indeed expected to derive it from there, according to the threshold of the policy, unless overriden in the local config.